src/EventSubscriber/LinkTypePlusEventSubscriber.php

```diff
@@ -0,0 +1,173 @@
+<?php
+
+namespace Drupal\commerce_gmo_linktypeplus\EventSubscriber;
+
+use Drupal\commerce_gmo_linktypeplus\Event\LinkTypePlusEvent;
+use Drupal\commerce_order\Entity\Order;
+use Drupal\Core\Entity\EntityTypeManagerInterface;
+use Drupal\Core\Logger\LoggerChannelFactoryInterface;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+
+/**
+ * Orderstatus change event subscriber.
+ */
+class LinkTypePlusEventSubscriber implements EventSubscriberInterface {
+
+  /**
+   * The entity type manager.
+   *
+   * @var \Drupal\Core\Entity\EntityTypeManagerInterface
+   */
+  protected $entityTypeManager;
+
+  /**
+   * Logger Factory.
+   *
+   * @var \Drupal\Core\Logger\LoggerChannelFactoryInterface
+   */
+  protected $loggerFactory;
+
+  /**
+   * Constructor.
+   *
+   * @param \Drupal\Core\Entity\EntityTypeManagerInterface $entityTypeManager
+   *   The entity type manager.
+   * @param \Drupal\Core\Logger\LoggerChannelFactoryInterface $loggerFactory
+   *   Logger .
+   */
+  public function __construct(EntityTypeManagerInterface $entityTypeManager, LoggerChannelFactoryInterface $loggerFactory) {
+    $this->entityTypeManager = $entityTypeManager;
+    $this->loggerFactory = $loggerFactory->get('linktype_event_subscriber');
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function getSubscribedEvents() {
+    return [
+      'Credit' => 'onlinkTypePaymentEvent',
```

Comment:
This section needs some improvement I think. While notifying us, gmo status varies based on the payment method. For eg: If we have used creditcard, the gmo status will be
 AUTH: Authorization
CAPTURE: Instant capture
UNPROSESSED: Not charged (payment is not completed due to authorization error etc.)
AUTHENTICATED: Not charged (3D registered)

Refer : https://docs.mul-pay.jp/linkplus/payment/credit/interface


If we use paypay the gmo status will be 

AUTH: Provisional sales
SALES: Actual
sales CAPTURE: Immediate sales
CANCEL: Cancellation
RETURN: Refund
PAYFAIL: Payment failure (a state where the payment failed due to an error)


Similarly the gmo status will be going to vary for other payments methods . As of now, we can't consolidate all the payment method's gmo status to map with drupal.

So if it is creditcard v can map it now. In future if they want for other methods they can extend it. This we can discuss and check